scrub_hub_project/authenticate/views.py

```diff
@@ -0,0 +1,67 @@
+from django.shortcuts import render
+
+# Create your views here.
+import json
+from django.contrib.auth import authenticate, login, logout
+from django.http import JsonResponse
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import require_POST
+from django.contrib.auth.models import User
+# Create your views here.
+
+def login_view(request):
+    if request.method == "POST":
+        data = json.loads(request.body)
+        username = data.get("username")
+        password = data.get("password")
+        
+        if username is None or password is None:
+            return JsonResponse({"detail":"Please provide username and password"})
+        user = authenticate(username=username, password=password)
+        if user is None:
+            return JsonResponse({"detail":"Invalid credentials"}, status=400)
+        login(request, user)
+        return JsonResponse({"detail": "Succesfully logged in!"})
+    elif request.method == "GET":
+        return render(request, 'authenticate/login.html')
+
+
+def logout_view(request):
+    if not request.user.is_authenticated:
+        return JsonResponse({"detail":"You are not logged in!"}, status=400)
+    logout(request)
+    return JsonResponse({"detail":"Succesfully logged out!"})
+
+
+@ensure_csrf_cookie
+def session_view(request):
+    if not request.user.is_authenticated:
+        return JsonResponse({"isAuthenticated": False})
+    return JsonResponse({"isAuthenticated": True})
+
+def dashboard_view(request):
+    if not request.user.is_authenticated:
+        # return JsonResponse({"isAuthenticated": False})
+        return render(request, 'authenticate/login.html')
+    context = { 'username': request.user.username }
+    return render(request, 'authenticate/dashboard.html', context)
+
+def register_view(request):
+    if request.method == "POST":
+        data = json.loads(request.body)
```

Comment:
```suggestion

@ensure_csrf_cookie
def register_view(request):
    if request.method == "POST":
        data = json.loads(request.body)
```
Make sure the user will have a CSRF token when they visit a page from which a POST request will be made.